src/ore/src/option.rs

```diff
@@ -45,3 +46,69 @@ impl<T> OptionExt<T> for Option<T> {
         self.as_ref().map(|x| x.deref().to_owned())
     }
 }
+
+/// Extension methods for things, like `[Option]`, that are displayable if a default value is supplied.
+pub trait OrDisplay<D> {
```

Comment:
Hm, I was imagining the trait returning a concrete type always. See for example this extension trait on iterators: https://docs.rs/itertools/0.10.0/itertools/trait.Itertools.html

so this would be:

```rust
trait OrDisplay {
    fn or_display(self, default: D) -> OrDisplay<Self, D>;
    fn or_display_with(self, fn: F) -> OrDisplayWith<Self, F>;
}
```

Reply:
In this case it would only be possible to implement it for `Option`, right?